include/picongpu/particles/manipulators/unary/SparserMacroParticlesPerSuperCell.hpp

```diff
@@ -0,0 +1,90 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+//! @file defines sparser functor for keeping only macro particles from one cell per superCell
+
+#pragma once
+
+#include "picongpu/simulation_defines.hpp"
+
+#include <cstdint>
+#include <string_view>
+
+namespace picongpu::particles::manipulators::unary::acc
+{
+    namespace detail
+    {
+        struct SparserParam
+        {
+            //! local index of cell in each superCell whose macro particles to keep
+            using idxCellToKeep = pmacc::math::CT::shrinkTo<pmacc::math::CT::Int<1, 1, 1>, picongpu::simDim>::type;
+        };
+    } // namespace detail
+
+
+    /** functor marking for deletion all passed macro particles which are not in a specified cell of a superCell
+     *
+     * to be passed as template parameter to picongpu::particles::manipulators::unary::FreeTotalCellOffset
+     *
+     * @tparam T_SparserParam which in superCell cellIdx to keep macro particles
+     *
+     * @attention any application of this functor to a species must be followed by FillAllGaps<SpeciesName> to
+     *  preserve particle data integrity!
+     */
+    template<typename T_SparserParam = detail::SparserParam>
+    struct SparserMacroParticlesPerSuperCell
+    {
+        template<typename T_Particle>
+        HDINLINE void operator()(
+            DataSpace<picongpu::simDim> const& particleOffsetInCellsToTotalOrigin,
+            T_Particle& particle)
+        {
+            DataSpace<picongpu::simDim> const superCellSize_RT = picongpu::SuperCellSize::toRT();
+            DataSpace<picongpu::simDim> const idxCellToKeep_RT = T_SparserParam::idxCellToKeep::toRT();
+
+            // is particle in cell whose particles we want to keep?
+            bool keepParticle = true;
+            for(uint8_t i = 0; i < picongpu::simDim; ++i)
+            {
+                keepParticle = keepParticle
+                    && ((particleOffsetInCellsToTotalOrigin[i] % superCellSize_RT[i]) == idxCellToKeep_RT[i]);
+            }
+
+            if(!keepParticle)
+            {
+                // mark particle as invalid
+                particle[multiMask_] = 0;
+                return;
+            }
+            else
+            {
+                constexpr bool hasWeighting
+                    = pmacc::traits::HasIdentifier<typename T_Particle::FrameType, weighting>::type::value;
+
+                if constexpr(hasWeighting)
+                    // update  weight to approximately maintain total density
+                    particle[weighting_] *= pmacc::math::CT::volume<typename picongpu::SuperCellSize>::type::value;
+            }
+        }
+
+        static constexpr std::string_view name = "SparserMacroParticlesPerSuperCell";
```

Comment:
Just a question: why are you using a `std::string_view` here instead of a `std::string`?

Reply:
I think I just followed your previous implementation, you used a string_view so I kept it a string view

Reply:
Nope, it wasn't your previous implementation, I don't remember, well it's fixed now